Lib/test/test_asyncio/test_server.py

```diff
@@ -212,6 +214,9 @@ async def serve(rd, wr):
         await asyncio.sleep(0)
         self.assertTrue(task.done())
 
+        with self.assertRaisesRegex(RuntimeError, r'is closed'):
```

Comment:
Please add new tests for the state machine, it would be nice to check if this addresses https://github.com/python/cpython/issues/123720 and similar issue and add regression tests for them